src/hooks/useStakingValidator.ts

```diff
@@ -1,32 +1,53 @@
 import { useState } from 'react';
 
+import { BonsaiHooks } from '@/bonsai/ontology';
 import {
   BondStatus,
   Validator,
 } from '@dydxprotocol/v4-proto/src/codegen/cosmos/staking/v1beta1/staking';
 import { useQuery } from '@tanstack/react-query';
 import { groupBy } from 'lodash';
-import { shallowEqual } from 'react-redux';
 
 import { ENVIRONMENT_CONFIG_MAP } from '@/constants/networks';
+import { EMPTY_ARR } from '@/constants/objects';
 
 import { useDydxClient } from '@/hooks/useDydxClient';
 
-import { calculateSortedUnbondingDelegations } from '@/state/accountCalculators';
-import { getStakingDelegations } from '@/state/accountSelectors';
+import { appQueryClient } from '@/state/appQueryClient';
 import { getSelectedNetwork } from '@/state/appSelectors';
 import { useAppSelector } from '@/state/appTypes';
 
 import { MustBigNumber } from '@/lib/numbers';
 
+export const refreshStakingData = () => {
+  appQueryClient.invalidateQueries({
+    queryKey: ['validator', 'staking'],
+  });
+  appQueryClient.invalidateQueries({
+    queryKey: ['validator', 'accountBalances'],
+  });
+};
+
+export const useSortedUnbondingDelegations = () => {
+  const unbondingDelegations = BonsaiHooks.useUnbondingDelegations().data;
+  if (unbondingDelegations?.length) {
+    const sortedUnbondingDelegations = [...unbondingDelegations];
+    sortedUnbondingDelegations.sort(
+      (a, b) => new Date(a.completionTime).getTime() - new Date(b.completionTime).getTime()
+    );
```

Comment:
can shorten and make the `sortedUnbondingDelegations` var more accurate

`[...unbondingDelegations].sort(...)`